tests/kernel/mem_protect/obj_validation/src/main.c

```diff
@@ -112,10 +121,109 @@ void test_generic_object(void)
 	}
 }
 
+/**
+ * @brief Test requestor thread will implicitly be assigned permission on the
+ * dynamically allocated object
+ *
+ * @details
+ * - Create kernel object semaphore, dynamically allocate it from the calling
+ *   thread's resource pool.
+ * - Check that object's address is in bounds of that memory pool.
+ * - Then check the requestor thread will implicitly be assigned permission on
+ *   the allocated object by using semaphore API k_sem_init()
+ *
+ * @ingroup kernel_memprotect_tests
+ *
+ * @see k_object_alloc()
+ */
+void test_kobj_assign_perms_on_alloc_obj(void)
+{
+	test_dyn_sem0 = k_object_alloc(K_OBJ_SEM);
+	zassert_not_null(test_dyn_sem0, "Cannot allocate sem k_object");
+
+	struct k_thread *thread = _current;
+
+	intptr_t sem_addr = (intptr_t)&test_dyn_sem0;
+	intptr_t start_addr = (intptr_t)&thread->resource_pool;
+	size_t size_heap = CONFIG_HEAP_MEM_POOL_SIZE;
+
+	/* check that kernel object may be dynamically allocated,
+	 * drawing memory from the calling thread's memory pool
+	 */
+#if defined(CONFIG_X86)
```

Comment:
I don't understand why we have different cases here. resource pools are not implemented in an arch-specific manner.